### PR TITLE
Renaming class MasterSlave to IntraComm and corresponding references

### PR DESCRIPTION
**Pull request by IshaanDesai**
## Main changes of this PR

This PR is a follow-up of https://github.com/precice/precice/pull/1253 and completes the renaming process. In particular the renaming of the class `MasterSlave` to `IntraComm` is done, along with the cascading name changes that are required.

## Author's checklist

* [x] I added a changelog file with `make changelog` if there are user-observable changes since the last release.
* [x] I ran `make format` to ensure everything is formatted correctly.
* [x] I sticked to C++14 features.
* [x] I sticked to CMake version 3.16.3.
* [ ] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [ ] Does the changelog entry make sense? Is it formatted correctly?
* [ ] Do you understand the code changes?

### CODE REVIEW

File: src/precice/config/ParticipantConfiguration.cpp [hunk 1]
```diff
@@ -194,65 +194,66 @@ ParticipantConfiguration::ParticipantConfiguration(
...
-    doc = "A solver in parallel needs a communication between its ranks. ";
-    doc += "By default (which is this option), the participant's MPI_COM_WORLD is reused.";
-    doc += "This tag is only used to ensure backwards compatibility.";
-    tagMaster.setDocumentation(doc);
-
-    intraCommTags.push_back(tagMaster);
-  }
-  for (XMLTag &tagMaster : intraCommTags) {
-    tag.addSubtag(tagMaster);
+  for (std::string tag_name : {TAG_MASTER, TAG_PRIMARY_RANK}) {
```
Comment (uekerman) [inline src/precice/config/ParticipantConfiguration.cpp:197]: Just a reminder: please open an issue such that we remove the `TAG_MASTER` here with v3.0.

File: src/precice/config/ParticipantConfiguration.cpp [hunk 2]
```diff
@@ -194,65 +194,66 @@ ParticipantConfiguration::ParticipantConfiguration(
...
+    {
+      XMLTag tagPrimary(*this, "sockets", intraCommOcc, tag_name);
+      doc = "A solver in parallel needs a communication between its ranks. ";
+      doc += "By default, the participant's MPI_COM_WORLD is reused.";
+      doc += "Use this tag to use TCP/IP sockets instead.";
+      tagPrimary.setDocumentation(doc);
```
Comment (uekerman) [inline src/precice/config/ParticipantConfiguration.cpp:203]: This tag should be called "tagIntraComm".

File: src/precice/config/ParticipantConfiguration.hpp [hunk 3]
```diff
@@ -77,6 +77,7 @@ class ParticipantConfiguration : public xml::XMLTag::Listener {
   const std::string TAG_WATCH_INTEGRAL = "watch-integral";
   const std::string TAG_WATCH_POINT    = "watch-point";
   const std::string TAG_MASTER         = "master";
+  const std::string TAG_PRIMARY_RANK   = "primary-rank";
```
Comment (uekerman) [inline src/precice/config/ParticipantConfiguration.hpp:80]:
```suggestion
  const std::string TAG_INTRA_COMM   = "intra-comm";
```

File: src/utils/Parallel.cpp [hunk 4]
```diff
@@ -304,7 +304,7 @@ void Parallel::splitCommunicator(const std::string &groupName)
         accessorGroups[groupMap[name]].size++;
       }
     }
-    // Step 2 send groupCount to Master
+    // Step 2 send groupCount to Primary
```
Comment (uekerman) [inline src/utils/Parallel.cpp:307]:
```suggestion
    // Step 2 send groupCount to primary rank
```
similar comments below

**Review by uekerman — CHANGES_REQUESTED** [review]
Skimmed over the changes. 
The tag should be `intra-comm`, not `primary-rank`. Everything else seems pretty straight forward.

**Comment by IshaanDesai** [comment]
@fsimonis your review I need just to verify whether the optional tags have been properly set. I ran the tests with the old flags and the relevant deprecated warnings are seen.

**Comment by fsimonis** [comment]
_For the record_ @IshaanDesai I would advise to keep some of the old integration tests and copy them into a `deprecated` test suite. This allows the CI to make sure that everything still works as expected.

**Review by fsimonis — CHANGES_REQUESTED** [review]
Found one naming mistake, which should also impact tests.

File: docs/changelog/1274.md [hunk 5]
```diff
@@ -0,0 +1 @@
+- Renamed class 'MasterSlave' to 'IntraComm' and changed all corresponding names. Changed the XML tag 'master' to 'intra-comm' and changed the XML attribute 'on-slaves' to 'on-secondary-rank'.
```
Comment (fsimonis) [inline docs/changelog/1274.md:1]: Maybe we should mention backwards compatibility here.

**Review by fsimonis — APPROVED** [review]
The changes are in a good shape now. Just one hint regarding the changelog entry and then we can merge.

File: docs/man/man1/precice-tools.1 [hunk 6]
```diff
@@ -34,7 +34,7 @@ Prints the preCICE version string ( \fIprecice::getVersionInformation()\fR ) to
 
 .TP
 .B check \fIfile\fR [\fIparticipant\fR [\fIsize\fR]]
-Checks given preCICE configuration \fIfile\fR and prints any errors. Pass the name of a \fIparticipant\fR to fully check its configuration. The optional \fIsize\fR indicates how many ranks run in parallel, which checks custom \fI<master: ... />\fR tags.
+Checks given preCICE configuration \fIfile\fR and prints any errors. Pass the name of a \fIparticipant\fR to fully check its configuration. The optional \fIsize\fR indicates how many ranks run in parallel, which checks custom \fI<primary-rank: ... />\fR tags.
```
Comment (uekerman) [inline docs/man/man1/precice-tools.1:37]: This still looks wrong

**Review by uekerman — APPROVED** [review]
Only had a quick look again.

File: docs/changelog/1274.md [hunk 7]
```diff
@@ -1 +1 @@
-- Renamed class 'MasterSlave' to 'IntraComm' and changed all corresponding names. Changed the XML tag 'master' to 'intra-comm' and changed the XML attribute 'on-slaves' to 'on-secondary-rank'.
+- Renamed class 'MasterSlave' to 'IntraComm' and changed all corresponding names. Changed the XML tag 'master' to 'intra-comm' and changed the XML attributes 'on-slaves' and 'on-master' to 'on-secondary-rank' and 'on-primary-rank' respectively. The tag 'master' and attributes 'on-slaves' and 'on-master' are deprecated.
```
Comment (uekerman) [inline docs/changelog/1274.md:1]: `on-secondary-ranks`, right? not `on-secondary-rank`?

File: docs/man/man1/precice-tools.1 [hunk 8]
```diff
@@ -34,7 +34,7 @@ Prints the preCICE version string ( \fIprecice::getVersionInformation()\fR ) to
 
 .TP
 .B check \fIfile\fR [\fIparticipant\fR [\fIsize\fR]]
-Checks given preCICE configuration \fIfile\fR and prints any errors. Pass the name of a \fIparticipant\fR to fully check its configuration. The optional \fIsize\fR indicates how many ranks run in parallel, which checks custom \fI<primary-rank: ... />\fR tags.
+Checks given preCICE configuration \fIfile\fR and prints any errors. Pass the name of a \fIparticipant\fR to fully check its configuration. The optional \fIsize\fR indicates how many ranks run in parallel, which checks custom \fI<on-primary-rank: ... />\fR tags.
```
Comment (uekerman) [inline docs/man/man1/precice-tools.1:37]: `intra-comm`, not `on-primary-rank`

File: docs/changelog/1274.md (hunk 7 above)
Comment (IshaanDesai) [reply docs/changelog/1274.md:1]: Fixed in https://github.com/precice/precice/pull/1283

File: docs/man/man1/precice-tools.1 (hunk 8 above)
Comment (IshaanDesai) [reply docs/man/man1/precice-tools.1:37]: Fixed in https://github.com/precice/precice/pull/1283

**Pull request merged**